bio.go

```diff
@@ -137,30 +149,40 @@ func (b *writeBio) WriteTo(w io.Writer) (rv int64, err error) {
 
 func (self *writeBio) Disconnect(b *C.BIO) {
 	if loadWritePtr(b) == self {
-		writeBioMapping.Del(token(C.X_BIO_get_data(b)))
 		C.X_BIO_set_data(b, nil)
 	}
 }
 
-func (b *writeBio) MakeCBIO() *C.BIO {
+func (b *writeBio) makeCBIO() *C.BIO {
 	rv := C.X_BIO_new_write_bio()
-	token := writeBioMapping.Add(unsafe.Pointer(b))
-	C.X_BIO_set_data(rv, unsafe.Pointer(token))
+	C.X_BIO_set_data(rv, unsafe.Pointer(b))
 	return rv
 }
 
-var readBioMapping = newMapping()
-
 type readBio struct {
 	data_mtx        sync.Mutex
 	op_mtx          sync.Mutex
 	buf             []byte
 	eof             bool
 	release_buffers bool
+	cBIO            *C.BIO
+}
+
+func newReadBIO() *readBio {
+	ret := &readBio{}
+	ret.cBIO = ret.makeCBIO()
```

Comment:
The one place where this is used, this `BIO` is passed to `SSL_set_bio` which transfers ownership of this `BIO` to the `SSL`, making it automatically freed when the SSL struct is.

Reply:
right, I should have read the man:
```
This function transfers ownership of rbio to ssl. It will be automatically freed
using BIO_free_all(3) when the ssl is freed.
```